compiler/resolution/wrappers.cpp

```diff
@@ -573,9 +576,19 @@ static void addArgCoercion(FnSymbol* fn, CallExpr* call, ArgSymbol* formal,
     castCall = NULL;
   }
 
-  if (castCall == NULL)
+  if (castCall == NULL) {
     // the common case
-    castCall = new CallExpr("_cast", fts, prevActual);
+    
+    //printf("HERE\n");
+    if( isDispatchParent(ats->type, fts->type) ) {
+      // Just use primitive cast to go from class type to object
+      castCall = new CallExpr(PRIM_CAST, fts, prevActual);
```

Comment:
Should this really be checking for object type? How do we cast a record to a sub-type?